### PR TITLE
BUG: Ensure data_columns is always a list (i.e. min_itemsize can exte…

### LINKED ISSUE #10381: BUG: DataFrame.to_hdf doesn't pass along min_itemsize for index
Unless I'm seeing something wrong

``` python
In [21]: df = DataFrame(dict(A = 'foo', B = 'bar'),index=range(5)).set_index("A")

In [22]: df.to_hdf('store.h5', 'test', format='table', min_itemsize={'index': 10})

In [23]: store = pd.HDFStore('store.h5')

In [24]: store.get_storer('test').table
Out[24]:
/test/table (Table(5,)) ''
  description := {
  "index": StringCol(itemsize=3, shape=(), dflt=b'', pos=0),
  "values_block_0": StringCol(itemsize=3, shape=(1,), dflt=b'', pos=1)}   # <---- I think this should be 10
  byteorder := 'irrelevant'
  chunkshape := (10922,)
  autoindex := True
  colindexes := {
    "index": Index(6, medium, shuffle, zlib(1)).is_csi=False}
```

and FYI this raises (not sure if it should work)

``` python
In [25]: df.index.name = 'theindex'

In [26]: df.to_hdf('store.h5', 'test2', format='table', min_itemsize={'theindex': 10})
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
```

Just a report right now... no time.

### PR DESCRIPTION
**Pull request by toobaz**
…nd it)

closes #10381

### CODE REVIEW

File: pandas/io/pytables.py [hunk 1]
```diff
@@ -3247,7 +3247,7 @@ def validate_data_columns(self, data_columns, min_itemsize):
         # evaluate the passed data_columns, True == use all columns
         # take only valide axis labels
         if data_columns is True:
-            data_columns = axis_labels
+            data_columns = list(axis_labels)
```
Comment (jreback) [inline pandas/io/pytables.py]: this type coercion will fail if you pass a string

Comment (jreback) [inline pandas/io/pytables.py]: in fact this should be not done here at all. The ``data_columns`` should be cleaned in a central location (IIRC they already are and just needs an adjustment)

File: pandas/io/tests/test_pytables.py [hunk 2]
```diff
@@ -1338,6 +1338,16 @@ def check_col(key, name, size):
                 [[124, 'abcdefqhij'], [346, 'abcdefghijklmnopqrtsuvwxyz']])
             self.assertRaises(ValueError, store.append, 'df_new', df_new)
 
+            # min_itemsize on Series with Multiindex (GH 10381)
```
Comment (jreback) [inline pandas/io/tests/test_pytables.py]: put this with tests that are actually testing ``min_itemsize``

**Comment by jreback** [comment]
can you rebase/update

**Comment by jreback** [comment]
can you rebase / update & add whatsnew for 0.18.2

**Comment by jreback** [comment]
can you update according to comments

**Comment by jreback** [comment]
can you rebase / update?

**Comment by toobaz** [comment]
Yeah, I'm just trying to focus on [one PR](https://github.com/pydata/pandas/pull/13205) at a time to limit (my) confusion... but this will be the next.

**Comment by jreback** [comment]
sure no prob

**Comment by jreback** [comment]
can you rebase / update?

**Comment by toobaz** [comment]
Not next week, I fear, but yes the one after.

**Comment by jreback** [comment]
closing, but pls reopen / comment if you can continue

**Comment by toobaz** [comment]
See #14728 (sorry for the long wait)

**Pull request closed without merging**